handbook/engineering/web/goals.md

```diff
@@ -4,29 +4,44 @@
-   Combining this with writing a few smaller extensions (part of goal (3)) allows us to dogfood the experience and inform us where the platform is lacking.
-3. **Build out compelling use cases with the extensions platform.**
-   This includes writing more extensions ourselves, but also extending the extension API with more capabilities to enable more use cases.
-   We have a long list of ideas and use cases customers shared with us that we can solve and/or enable customers to solve with extensions.
-   Some of these integrate Sourcegraph with more external services to suit the setups of more customers, others provide completely new value no other product provides.
+_To reach our long-term goal, we set the following medium-term goals to guide our short-term iteration plans. These medium-term goals are listed in order of rought priority. This means we preference the sooner-listed goal when making progress on one would conflict with another goal, but it doesn't mean we only work on the first goal until it's done – we balance our iterations in sum so we can make some progress on all of these goals over a quarter. Within each goal, the sub-goals are also listed in (stricter) priority order._ 
+
+1. **Make the Sourcegraph web interfaces more consistent and improve discoverability of Sourcegraph features.**
+   1. Make progress on design debt throughought the web app where the web team owns the design: repo page areas and panels, navigation and headers, general theming, and the blog page (file viewer). 
+   1. Build and update designs to help users discover or understand everything Sourcegraph offers, like new navigation menus and header concepts. 
+
+1. **Make Sourcegraph extensions a core part of Sourcegraph users' experiences.**
+   1. Track anonymized, general usage of extensions to determine which extensions are most successful at adding value for our users to inform our future work.
+   1. Build new API endpoints that are robust and obviously useful (used immediately in an extension), to grow adoption of extensions. 
+   1. Make the extensions action bar clear and scalable, and help users discover and use our extensions. 
+   1. Improve discoverability and design of the extension registry to enable users to find the most useful extensions and make users excited to build their own extension. 
+
+1. **Make code insights an entirely new reason to use Sourcegraph.**
+   1. Focus on building prototypes to solve the most pressing needs of our customers, to clearly demonstrate the value of code insights. 
+   1. Expose more generalizable metrics that let our users measure and track their own goals, whether those are migrations, code smells, security needs, cross-collaboration, or other information about code. 
+   1. Expose features that let our users build their own insights. 
```

Comment:
To what extent does this goal overlap with the goal of making Sourcegraph more extensible? 

Will the extensions platform have shared ownership once we have separate extensibility and code insights teams, in the same way that the web app is shared between multiple teams? My understanding right now is that these features would be added to the extension API. Maybe this is only a technical/implementation conflict, since code insights and extensions are sufficiently distinct conceptually to create separate teams.

Reply:
It definitely overlaps some. I think this is more of an eng team organization question (who owns it, where the features go in the codebase) and I will leave that to @jeanduplessis!

If you're also trying to ask "why is this not covered in the extensions goal" I think it's because insights may go beyond extensions. Not all code insights features will necessarily be delivered as extensions – even if they use the extensions API, we might build them into Sourcegraph directly, or build entirely new UX flows around this (rather than "enable the extension in the registry, set it up, etc.). 

Moreover, nothing in the extensions medium-term goal talks about building new extensions, just new endpoints as we need them. We just built some AWESOME extensions, and there's nothing on the brainstorm list of new extensions that is so pressing that I think it's valuable to prioritize that over all these other goals. So even if code insights was just "new extensions that are code insights!" that still wouldn't fit into the medium-term extensions goal now, because "build a new extension, even if it doesn't require a new endpoint" is not part of the extensions goal, but is part of the code insights goal.

Reply:
I can confirm having separate teams for code insights and extensions is purely related to organizational design to allow for better focus and growth. I would expect that a close collaboration between the two teams will be needed. This collaboration will primarily be facilitated by the teams' EMs and PMs.